src/fields/extend_p.rs

```diff
@@ -28,11 +37,15 @@ pub struct ExtendContext<FE> {
 /// * `evaluations` - The desired output length
 pub(super) fn extend_precompute<FE>(nodes_len: usize, evaluations: usize) -> ExtendContext<FE>
 where
-    FE: ProofFieldElement,
+    FE: NttFieldElement,
 {
-    let mut reciprocals = Vec::with_capacity(evaluations + 1);
+    let ntt_size = evaluations.next_power_of_two();
+
+    let reciprocals_len = cmp::max(evaluations + 1, ntt_size);
+    let mut reciprocals = Vec::with_capacity(reciprocals_len);
     reciprocals.push(FE::ZERO);
-    reciprocals.extend((1..=evaluations).map(|i| FE::from_u128(i.try_into().unwrap()).mul_inv()));
+    reciprocals
+        .extend((1..reciprocals_len).map(|i| FE::from_u128(i.try_into().unwrap()).mul_inv()));
```

Comment:
We should consider precomputing reciprocals at compile time. For a given instantiation of Longfellow (e.g. proving a specific attribute from an mDoc), the proof sizes will be fixed so it'll be the same reciprocals all the time. #103